robot:
  # Publish all joint states -----------------------------------

  joint_state_controller:
    type: joint_state_controller/JointStateController
    publish_rate: 50

  # Sawyer SDK Controllers: Position --------------------------
  right_joint_position_controller:
    type: sawyer_sim_controllers/SawyerPositionController
    topic_joint_command: /robot/limb/right/joint_command
    topic_set_speed_ratio: /robot/limb/right/set_speed_ratio
    joints:
      right_j0_controller:
        joint: right_j0
        pid: {p: 700,  i: 0.01, d: 100}
      right_j1_controller:
        joint: right_j1
        pid: {p: 10000,  i: 100, d: 100}
      right_j2_controller:
        joint: right_j2
        pid: {p: 10000,  i: 100, d: 100}
      right_j3_controller:
        joint: right_j3
        pid: {p: 10000,  i: 100, d: 100}
      right_j4_controller:
        joint: right_j4
        pid: {p: 10000,  i: 100, d: 100}
      right_j5_controller:
        joint: right_j5
        pid: {p: 10000,  i: 100, d: 100}
      right_j6_controller:
        joint: right_j6
        pid: {p: 10000,  i: 100, d: 100}

  # Sawyer SDK Controllers: Velocity --------------------------
  right_joint_velocity_controller:
    type: sawyer_sim_controllers/SawyerVelocityController
    topic: /robot/limb/right/joint_command
    joints:
      right_j0_controller:
        joint: right_j0
        pid: {p: 700,  i: 0.01, d: 100}
      right_j1_controller:
        joint: right_j1
        pid: {p: 100,  i: 100, d: 100}
      right_j2_controller:
        joint: right_j2
        pid: {p: 100,  i: 100, d: 100}
      right_j3_controller:
        joint: right_j3
        pid: {p: 100,  i: 100, d: 100}
      right_j4_controller:
        joint: right_j4
        pid: {p: 100,  i: 100, d: 100}
      right_j5_controller:
        joint: right_j5
        pid: {p: 100,  i: 100, d: 100}
      right_j6_controller:
        joint: right_j6
        pid: {p: 100,  i: 100, d: 100}

  # Sawyer SDK Controllers: Effort --------------------------
  right_joint_effort_controller:
    type: sawyer_sim_controllers/SawyerEffortController
    topic: /robot/limb/right/joint_command
    joints:
      right_j0_controller:
        joint: right_j0
        pid: {p: 700,  i: 0.01, d: 100}
      right_j1_controller:
        joint: right_j1
        pid: {p: 100,  i: 100, d: 100}
      right_j2_controller:
        joint: right_j2
        pid: {p: 100,  i: 100, d: 100}
      right_j3_controller:
        joint: right_j3
        pid: {p: 100,  i: 100, d: 100}
      right_j4_controller:
        joint: right_j4
        pid: {p: 100,  i: 100, d: 100}
      right_j5_controller:
        joint: right_j5
        pid: {p: 100,  i: 100, d: 100}
      right_j6_controller:
        joint: right_j6
        pid: {p: 100,  i: 100, d: 100}

  # Sawyer SDK Controllers: Gravity Compensation ------------
  right_joint_gravity_controller:
    type: sawyer_sim_controllers/SawyerGravityController
    topic: /robot/limb/right/gravity_command
    joints:
      right_j0_controller:
        joint: right_j0
        pid: {p: 700,  i: 0.01, d: 100}
      right_j1_controller:
        joint: right_j1
        pid: {p: 100,  i: 100, d: 100}
      right_j2_controller:
        joint: right_j2
        pid: {p: 100,  i: 100, d: 100}
      right_j3_controller:
        joint: right_j3
        pid: {p: 100,  i: 100, d: 100}
      right_j4_controller:
        joint: right_j4
        pid: {p: 100,  i: 100, d: 100}
      right_j5_controller:
        joint: right_j5
        pid: {p: 100,  i: 100, d: 100}
      right_j6_controller:
        joint: right_j6
        pid: {p: 100,  i: 100, d: 100}

